extension/src/background/messageListener/popupMessageListener.ts

```diff
@@ -917,6 +917,24 @@ export const popupMessageListener = (request: Request) => {
     return { error: "Session timed out" };
   };
 
+  const signFreighterSorobanTransaction = () => {
+    const { transaction, network } = request;
+
+    const transaction2 = SorobanSdk.TransactionBuilder.fromXDR(
+      transaction,
+      network,
+    );
+
+    const privateKey = privateKeySelector(store.getState());
+    if (privateKey.length) {
+      const sourceKeys = SorobanSdk.Keypair.fromSecret(privateKey);
+      transaction2.sign(sourceKeys);
+      return { signedTransaction: transaction2.toXDR() };
```

Comment:
@piyalbasu as expected, encoding the signed tx to XDR was the last thing needed to make this work. So now you can successfully send some amount of a Soroban token! 🎉  (you should have a few FTT from my tests).